hw_07/hw_07/HW.07.Task2/Program.cs

```diff
@@ -0,0 +1,170 @@
+                if (input.Contains(div2)) input = input.Replace(div2, div); //Removes duplicate spaces.
+            }
+            while (!input.Contains(div));
+
+            DeleteLongestWord(input);
+
+            ChangeLongestShortestWord(input);
+
+            CountLettersPunctuation(input);
+
+            SortWordsDescend(input);
+
+            EndApplication();
+        }
+        static void ContinueVerify()
+        {
+            string? input2 = Console.ReadLine();
+            //bool result2 = char.TryParse(input2, out char yn);
+            //if (result2 == true & yn == 'Y' || yn == 'y')
+            if (input2.Equals("Y", StringComparison.OrdinalIgnoreCase))
+                Console.WriteLine($"Input line with correct dividing symbols.");
+            else EndApplication();
+        }
+        static void DeleteLongestWord(string text)
+        {
+            string[] lines = text.Split(' ');
+            string longest = lines[0];
+
+            for (int i = 1; i < lines.Length; i++)
+            {
+                if (longest.Length < lines[i].Length) longest = lines[i];
+            }
+            text = text.Replace(longest, "");
+            Console.WriteLine("The longest word is deleted: " + text);
+        }
+        static void ChangeLongestShortestWord(string text)
+        {
+            string[] lines = text.Split(' ');
+            string longest = lines[0];
+            string shortest = lines[0];
+            string temp;
+            int ind_l = 0, ind_sh = 0;
+
+            for (int i = 0; i < lines.Length; i++)
+            {
+                if (longest.Length < lines[i].Length)
+                {
+                    longest = lines[i]; ind_l = i;
+                }
+                if (shortest.Length > lines[i].Length)
+                {
+                    shortest = lines[i]; ind_sh = i;
+                }
+            }
+
+            //Changes positions of the shortest and the longest words.
+            temp = lines[ind_l];
+            lines[ind_l] = lines[ind_sh];
+            lines[ind_sh] = temp;
+
+            Console.WriteLine("The positions of the shortest and the longest words are changed.");
+
+            foreach (string str in lines)
+            {
+                Console.Write(str + " ");
+            }
+            Console.WriteLine();
+        }
+        static void CountLettersPunctuation(string text)
+        {
+            int count_let = 0;
+            int count_punct = 0;
+            int symb;
+
+            foreach (char ch in text)
+            {
+                symb = (char)ch;
+                switch (symb)
+                {
+                    case >= 65 when (symb<= 90):
+                        count_let++; break;
+                    case >= 97 when (symb <= 122):
+                        count_let++; break;
+                    case 33:
+                    case 34:
+                    case 58:
+                    case 59:
+                    case 63:
```

Comment:
what are there magic numbers (33, 34, 58)?
you can use special char methods, such as Char.IsNumber(), etc.